exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/WebServer.java

```diff
@@ -219,9 +220,11 @@ private ServletContextHandler createServletContextHandler(final boolean authEnab
       servletContextHandler.addServlet(dynamicHolder, "/dynamic/*");
     }
 
+    final Map<String, String> responseHeaders = ResponseHeadersSettingFilter.retrieveResponseHeaders(config);
     if (authEnabled) {
       //DrillSecurityHandler is used to support SPNEGO and FORM authentication together
-      servletContextHandler.setSecurityHandler(new DrillHttpSecurityHandlerProvider(config, workManager.getContext()));
+      servletContextHandler.setSecurityHandler(new DrillHttpSecurityHandlerProvider(config, workManager.getContext(),
+          (req, resp) -> responseHeaders.forEach(resp::setHeader)));
```

Comment:
Looks like the response is used here only. Is it possible to use and pass `Consumer` instead of `BiConsumer`?